### PR TITLE
Replace u_buy_item with u_spawn_item

### LINKED ISSUE #51668: Cannot acquire quest items if owing money to quest giver
### Describe the bug

For missions that give you equipment via `u_buy_item`, if you owe money to the questgiver this will pop up a trade window. If you cancel out of that (because you don't know what you're supposed to do) you get a message "You can't afford it" and don't receive the item, which can make the mission impossible to complete.
This also applies to quest rewards, anything scripted to use `u_buy_item`.

### Steps To Reproduce

1. Start in evac shelter, use computer, teleport to Refugee Center.
2. Talk to merchant, get quest to deliver data to Hub01, teleport there and hand in the quest.
3. Accept the first mission "Return Field Data", complete it, then choose the trade option and buy something cheap on credit (without paying)
4. Ask about other missions and accept the "Steal a dead man's mind" quest.
5. Instead of receiving the quest items necessary to complete it you get an error "You can't afford it"
6. The mission cannot be completed.

### Expected behavior

Free quest items and quest rewards should be given to you regardless of debt status.

### Screenshots

image

image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: f10a1766 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by eltank**
#### Summary
Bugfixes "Replace u_buy_item with u_spawn_item"

#### Purpose of change

Fixes #51668
`u_buy_item` is used extensively in mission dialogue scripts to give free items to the player, either at the start of a mission or on completion. However, if the player owes money to the questgiver this pops up a trade window and canceling that results in no item being given.

#### Describe the solution

Replace `u_buy_item` with the recently-implemented `u_spawn_item` (#51603), which gives you the item unconditionally.
The `u_spawn_item` handler `talk_effect_fun_t::set_u_spawn_item` was missing feedback messages (like "X gives you Y"), so I added that and then refactored the duplicated code between `set_u_spawn_item` and `set_u_buy_item`.
I made the feedback message conditional on the presence of a beta talker, so that `u_spawn_item` can still be used outside of dialogue (e.g. EOC).
Added JSON validation to prevent `u_buy_item` from being used without specifying a positive `cost` value.
Fixed `u_buy_monster`, which has a similar problem, by adding a cost check in the handler (3rd commit).

#### Describe alternatives you've considered

I thought about also renaming `u_spawn_item` to `u_receive_item` to make it more descriptive but I didn't want to cause more churn.

I considered fixing `u_buy_item` by fixing the cost check (what I did for `u_buy_monster`), but I also wanted to differentiate it from `u_spawn_item` and get rid of code duplication.

#### Testing

Follow the instructions from the bug. Accept the "Steal a dead man's mind" quest while owing money and you'll still get the mind splicer:

image

image

Without the fix:

image

#### Additional context

The only remaining instances of `u_buy_item`:
```
    "id": "TALK_RANCH_BARKEEP_TAP",
    "type": "talk_topic",
    "dynamic_line": "Our selection is a bit limited at the moment.",
    "responses": [
      {
        "text": "[$8] I'll take a beer",
        "topic": "TALK_DONE",
        "effect": { "u_buy_item": "beer", "container": "bottle_glass", "count": 2, "cost": 800 }
      },
      {
        "text": "[$10] I'll take a shot of brandy",
        "topic": "TALK_DONE",
        "effect": { "u_buy_item": "brandy", "container": "bottle_glass", "cost": 1000 }
      },
      {
        "text": "[$10] I'll take a shot of rum",
        "topic": "TALK_DONE",
        "effect": { "u_buy_item": "rum", "container": "bottle_glass", "cost": 1000 }
      },
      {
        "text": "[$12] I'll take a shot of whiskey",
        "topic": "TALK_DONE",
        "effect": { "u_buy_item": "whiskey", "container": "bottle_glass", "cost": 1200 }
      },
```

### CODE REVIEW

**Comment by eltank** [comment]
Broke `npc_talk_test`, fixing...

File: data/json/npcs/isherwood_farm/NPC_Eddie_Isherwood.json [hunk 1]
```diff
@@ -268,9 +271,9 @@
       "opinion": { "trust": 2, "value": 2 },
       "effect": [
         { "u_add_var": "u_did_eddie_missions", "type": "general", "context": "mission", "value": "yes" },
-        { "u_buy_item": "rope_6", "count": 1 },
+        { "u_spawn_item": "rope_6", "count": 1 },
         { "u_buy_monster": "mon_cow", "count": 1, "name": "Daisy" },
```
Comment (cainiaowu) [inline data/json/npcs/isherwood_farm/NPC_Eddie_Isherwood.json:275]: Wild guess, this "u_buy_monster" may also have the same behavior like "u_buy_item"

Comment (eltank) [reply data/json/npcs/isherwood_farm/NPC_Eddie_Isherwood.json:275]: From looking at the code, yes. Even though it's free it'll fail if you owe money.

Comment (eltank) [reply data/json/npcs/isherwood_farm/NPC_Eddie_Isherwood.json:275]: Added a C++ fix for that.

**Review by Maleclypse — COMMENTED** [review]
Excellent documentation.  Thank you!

**Pull request merged**